File: LicensedApps/RegistryLicensedApplication/RuntimeRegistryLicense.cs
using System;
using System.ComponentModel;

namespace RegistryLicensedApplication
{
	/// <summary>
	/// Summary description for RuntimeRegistryLicense.
	/// </summary>
	public class RuntimeRegistryLicense : License 
	{
		private Type type;

		public RuntimeRegistryLicense(Type type) 
		{
			if ( type == null ) throw new NullReferenceException("The licensed type reference may not be null.");
			this.type = type;
		}

		public override string LicenseKey 
		{
			get 
			{
				// Simply return the application's GUID
				return type.GUID.ToString();
			}
		}
		public override void Dispose() 
		{
		}
	}
}

File: LicensedApps/VBRegistryLicensedApplication/RuntimeRegistryLicense.vb
Imports System
Imports System.ComponentModel

Public Class RuntimeRegistryLicense
    Inherits License

    Private type As type = Nothing

    Public Sub New(ByVal type As type)
        Me.type = type
    End Sub

    Public Overrides ReadOnly Property LicenseKey() As String
        Get
            Return type.GUID.ToString()
        End Get
    End Property

    Public Overrides Sub Dispose()

    End Sub

End Class
